File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        max_arr = [-1 for _ in range(len(arr) + 1)]
        
        for i in range(len(arr) - 1, 0, -1):
            max_arr[i] = max(arr[i], max_arr[i + 1])
        
        return max_arr[1:]

File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.rb
# @param {Integer[]} arr
# @return {Integer[]}
def replace_elements(arr)
    ans = []
    (0...arr.length - 1).each do |idx|
         ans << arr.slice(idx + 1, arr.length).max
    end
    ans << - 1
    ans
end